app/src/main/java/com/appdev/eateryblueandroid/ui/viewmodels/ProfileViewModel.kt

```diff
@@ -108,11 +127,19 @@ class ProfileViewModel : ViewModel() {
     }
 
     fun loginSuccess(sessionId: String) {
-        _display.value = Display.Login(authenticating = true, progress = 0.9f)
-        if (state.value is State.LoggingIn)
-            saveLoginInfo((state.value as State.LoggingIn).netid, (state.value as State.LoggingIn).password)
-        else
-            saveLoginInfo((state.value as State.AutoLoggingIn).netid, (state.value as State.AutoLoggingIn).password)
+        //Catch the edge case in which a user logs out before it loads, then the load occurs.
+        if (state.value == State.Empty) return
+
+        val isAutoLogin = _state.value is State.AutoLoggingIn
+
+        if (!isAutoLogin) {
+            _display.value = Display.Login(authenticating = true, progress = 0.9f)
+            saveLoginInfo(
+                (state.value as State.LoggingIn).netid,
+                (state.value as State.LoggingIn).password
+            )
+        }
+
         viewModelScope.launch {
             val res1 = GetApiService.getInstance().fetchUser(
                 GetApiService.generateUserBody(sessionId = sessionId)
```

Comment:
Doesn't necessarily have to be changed now but in the future consider putting a these API calls in a repository especially as Eatery Blue continues to grow. The UI layer (these viewmodels, screens) shouldn't directly interact with these calls as we should have repository class for each different type of data we handle in your app (like one for Users, Eatery, Account, etc) so they can be used across multiple screens, separation of concerns, etc.

See [here](https://developer.android.com/topic/architecture/data-layer?hl=en)

Reply:
Most recent commits changed Auto-login and Recent Searches to singleton Repositories. Thanks for the great suggestion, this new format is far better for Jetpack Compose and just readability in general.